### PR TITLE
Fix null src-vector's for struct and list writers. 

### LINKED ISSUE #2393: CCE with structs in outer-joins
The problem is probably symmetric with right-outer-join and likely more fundamental then just outer-joins. There seems to be a problem of casting the nil vectors to certain other arrow type vectors
```clj
(t/deftest test-left-outer-with-structs
  (t/is (= nil #_{:res [{{:a 12, :b 12, :c 2} 1, {:a 12, :b 12, :c 0} 1, {:a 0, :b nil, :c nil} 1}
                        {{:a 12, :b 12, :c 2} 1, {:a 100, :b 100, :c 3} 1, {:a 12, :b 12, :c 0} 1}]
                  :col-types '{a :i64, b [:union #{:null :i64}], c [:union #{:null :i64}]}}
           (-> (tu/query-ra [:left-outer-join '[{a b}]
                             [::tu/blocks
                              [[{:a 12}, {:a 0}]
                               [{:a 12}, {:a 100}]]]
                             [::tu/blocks
                              [[{:b 12, :c {:foo 1}}, {:b 2, :c {:foo 1}}]
                               [{:b 12, :c {:foo 2}}, {:b 100, :c {:foo 2}}]]]]
                            {:preserve-blocks? true, :with-col-types? true})
               (update :res (partial mapv frequencies))))))
```

kind of blocking xtdb/xtdb#2352

### PR DESCRIPTION
**Pull request by FiV0**
Fixes #2393

### CODE REVIEW

File: core/src/main/clojure/xtdb/vector/writer.clj [hunk 1]
```diff
@@ -60,7 +60,7 @@
       (.setTypeId parent-duv parent-pos type-id)
       (.setOffset parent-duv parent-pos pos)
 
-      ;; TODO (#252) do we still need this here?
+      ;; TODO (#2155) do we still need this here?
```
Comment (FiV0) [inline core/src/main/clojure/xtdb/vector/writer.clj:63]: I didn't quite understand how this relates to #2155.

File: core/src/main/clojure/xtdb/vector/writer.clj [hunk 2]
```diff
@@ -60,7 +60,7 @@
...
       (.setValueCount parent-duv (inc parent-pos))
```
Comment (FiV0) [inline core/src/main/clojure/xtdb/vector/writer.clj:64]: I did not quite understand how this relates to #2155.

Comment (jarohen) [reply core/src/main/clojure/xtdb/vector/writer.clj:64]: #2154?

**Review by jarohen — APPROVED** [review]
nice one @FiV0 :clap:

**Pull request merged**